pkg/networkservice/common/heal/client_test.go

```diff
@@ -0,0 +1,378 @@
+		close(notifier)
+		if h != nil {
+			return h(ctx, in, opts...)
+		}
+		return nil, nil
+	}
+}
+
+func TestHealClient_New(t *testing.T) {
+	suits := []struct {
+		name     string
+		events   []*networkservice.ConnectionEvent
+		reported map[string]*networkservice.Connection
+	}{
+		{
+			name: "initial state transfer",
+			events: []*networkservice.ConnectionEvent{
+				{
+					Type: networkservice.ConnectionEventType_INITIAL_STATE_TRANSFER,
+					Connections: map[string]*networkservice.Connection{
+						"conn-1": {Id: "conn-1", NetworkService: "ns-1"},
+						"conn-2": {Id: "conn-2", NetworkService: "ns-2"},
+					},
+				},
+			},
+			reported: map[string]*networkservice.Connection{
+				"conn-1": {Id: "conn-1", NetworkService: "ns-1"},
+				"conn-2": {Id: "conn-2", NetworkService: "ns-2"},
+			},
+		},
+		{
+			name: "update event",
+			events: []*networkservice.ConnectionEvent{
+				{
+					Type: networkservice.ConnectionEventType_INITIAL_STATE_TRANSFER,
+					Connections: map[string]*networkservice.Connection{
+						"conn-1": {Id: "conn-1", NetworkService: "ns-1"},
+						"conn-2": {Id: "conn-2", NetworkService: "ns-2"},
+					},
+				},
+				{
+					Type: networkservice.ConnectionEventType_UPDATE,
+					Connections: map[string]*networkservice.Connection{
+						"conn-1": {Id: "conn-1", NetworkService: "ns-1-upd"},
+					},
+				},
+				{
+					Type: networkservice.ConnectionEventType_UPDATE,
+					Connections: map[string]*networkservice.Connection{
+						"conn-2": {Id: "conn-2", NetworkService: "ns-2-upd"},
+					},
+				},
+			},
+			reported: map[string]*networkservice.Connection{
+				"conn-1": {Id: "conn-1", NetworkService: "ns-1-upd"},
+				"conn-2": {Id: "conn-2", NetworkService: "ns-2-upd"},
+			},
+		},
+		{
+			name: "delete event",
+			events: []*networkservice.ConnectionEvent{
+				{
+					Type: networkservice.ConnectionEventType_INITIAL_STATE_TRANSFER,
+					Connections: map[string]*networkservice.Connection{
+						"conn-1": {Id: "conn-1", NetworkService: "ns-1"},
+						"conn-2": {Id: "conn-2", NetworkService: "ns-2"},
+					},
+				},
+				{
+					Type: networkservice.ConnectionEventType_DELETE,
+					Connections: map[string]*networkservice.Connection{
+						"conn-1": {Id: "conn-1", NetworkService: "ns-1"},
+					},
+				},
+			},
+			reported: map[string]*networkservice.Connection{
+				"conn-2": {Id: "conn-2", NetworkService: "ns-2"},
+			},
+		},
+	}
+
+	for i := range suits {
+		s := suits[i]
+		logrus.Info(s.name)
+		ok := t.Run(s.name, func(t *testing.T) {
+			f, err := newFixture()
+			require.Nil(t, err)
+			defer f.close()
+
+			for _, event := range s.events {
+				err = f.serverStream.Send(event)
+				require.Nil(t, err)
+			}
+
+			cond := func() bool {
+				equal := func(m1, m2 map[string]*networkservice.Connection) bool {
+					for k, v := range m1 {
+						if v2, ok := m2[k]; !ok || !proto.Equal(v, v2) {
```

Comment:
Do we really need additional dependency github.com/gogo/protobuf for comparison here?
As I know for check equality the testify using DeepEquals based on reflect.
Can we just use here require.Equals instead of proto.Equal and get rid of dep github.com/gogo/protobuf  ?

Reply:
Fixed